src/crl.rs

```diff
@@ -54,6 +54,75 @@ pub struct CertRevocationList<'a> {
     pub crl_number: Option<&'a [u8]>,
 }
 
+/// Representation of a RFC 5280[^1] profile Certificate Revocation List (CRL) revoked certificate
+/// entry.
+///
+/// [^1]: <https://www.rfc-editor.org/rfc/rfc5280#section-5>
+pub struct RevokedCert<'a> {
+    /// Serial number of the revoked certificate.
+    pub serial_number: &'a [u8],
+
+    /// The date at which the CA processed the revocation.
+    pub revocation_date: Time,
+
+    /// Identifies the reason for the certificate revocation.
+    pub reason_code: Option<RevocationReason>,
+
+    /// Provides the date on which it is known or suspected that the private key was compromised or
+    /// that the certificate otherwise became invalid. This date may be earlier than the revocation
+    /// date which is the date at which the CA processed the revocation.
+    pub invalidity_date: Option<Time>,
+}
+
+/// Identifies the reason a certificate was revoked.
+/// See RFC 5280 §5.3.1[^1]
+///
+/// [^1] <https://www.rfc-editor.org/rfc/rfc5280#section-5.3.1>
+#[derive(Debug, Clone, Eq, PartialEq)]
+#[allow(missing_docs)] // Not much to add above the code name.
+pub enum RevocationReason {
+    /// Unspecified should not be used, and is instead assumed by the absence of a RevocationReason
+    /// extension.
+    Unspecified,
+    KeyCompromise,
+    CaCompromise,
+    AffiliationChanged,
+    Superseded,
+    CessationOfOperation,
+    CertificateHold,
+    /// RemoveFromCrl only appears in delta CRLs that are unsupported.
+    RemoveFromCrl,
+    PrivilegeWithdrawn,
+    AaCompromise,
+}
+
+pub struct RevokedCerts<'a> {
+    reader: untrusted::Reader<'a>,
+}
+
+impl<'a> Iterator for RevokedCerts<'a> {
+    type Item = Result<RevokedCert<'a>, Error>;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        if self.reader.at_end() {
```

Comment:
Style suggestion: this is the kind of thing I'd write as a match. Or, consider `(!self.reader.at_end()).then_some(|| parse_revoked_cert(&mut self.reader))`? Ahh, I guess our MSRV doesn't support that.

Reply:
(Submitted #62 to pull up the MSRV.)

Reply:
That's much nicer :sparkles: Implemented in my recent update.